### ISSUE TITLE
TearDownTest on Sub Test

### ISSUE THREAD

**Issue by fanadol** [stretchr/testify#975]
Hi, thank you for your great library.

I want to know, is there any way to tear down in suite after each sub test ?

Thank you

**Comment by jokeyrhyme** [comment]
Maybe https://godoc.org/github.com/stretchr/testify/suite#AfterTest or https://godoc.org/github.com/stretchr/testify/suite#TearDownTestSuite ?

**Comment by fanadol** [comment]
Hi @jokeyrhyme thank you for your answer. Unfortunately it doesn't work for me. 
Here is my code example, here i try to use `BeforeTest` intead of `AfterTest` but i guess it should work the same.

```
func (suite *ClientSuite) BeforeTest(suiteName, testName string) {
	suite.Counter++
}

func (suite *TS) TestSomething() {
	suite.Run("one", func() {
		assert.Equal(suite.T(), 1, suite.Counter)
	})

	suite.Run("second", func() {
		assert.Equal(suite.T(), 2, suite.Counter)
	})
}
```

The test for `second` is error, because the `BeforeTest` only run once. Am i miss something in my implementation ?

**Comment by boyan-soubachov** [comment]
`AfterTest()` would be the best one for your use-case. You could put individual test tear-down in it. Please let me know if this does not work for you or answer your question :)

**Comment by fanadol** [comment]
Hi @boyan-soubachov thank you for your answer. Unfortunately it also doesn't work.  It seems like `AfterTest()` and `BeforeTest()` only executed in each `Test`, not in each `Sub Test`.

**Comment by boyan-soubachov** [comment]
Looking at the code now, I think the issue is because you're using the receiver variant of `Run()` instead of the package-level variant. The receiver variant has none of the tear-down and set-up support. It's something we still need to either remove or consolidate into the package-level variant.

If you use the `suite.Run(t, your_suite)`, this should work where `suite` = the `stretchr/testify/suite` package.

**Comment by fanadol** [comment]
I see, so the only workaround is to change the implementation from `Sub Test` into `Test`. Thank you for your explanation.

**Comment by boyan-soubachov** [comment]
> I see, so the only workaround is to change the implementation from `Sub Test` into `Test`. Thank you for your explanation.

Yup. Basically use the package-level `Run()` function and inject your suite into it as one of the arguments.
